docs/content/develop/resource.md

```diff
@@ -341,6 +341,28 @@ Replace `String` in the field type with one of the following options:
       description: |
         MULTI_LINE_FIELD_DESCRIPTION
 ```
+{{< /tab >}}
+{{< tab "Map" >}}
+```yaml
+  - !ruby/object:Api::Type::Map
+    name: 'API_FIELD_NAME'
+    description: |
+      MULTILINE_FIELD_DESCRIPTION
+    key_name: KEY_NAME
+    key_description: |
+      MULTILINE_KEY_FIELD_DESCRIPTION
+    value_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::String
+          name: 'FIELD_NAME'
+          description: |
+            MULTI_LINE_FIELD_DESCRIPTION
+```
+
+This type is only used for string -> complex type mappings, use "KeyValuePairs" for simple mappings. Complex maps can't be represented natively in Terraform, and this type is transformed into an associative array (TypeSet) with the key merged into the object alongside other top-level fields.
```

Comment:
How do we handle set diff/update logic in the generated code? I could imagine that we could make the hash function only look at the key_name value

Reply:
Not how Terraform hash functions work, unfortunately- they don't pair elements in the set for field-by-field comparison, they determine the _entire_ set diff.

Effectively for a set with elements A, B, and C, the diff is {hash(A), hash(B), hash(C)} vs {hash(A'), hash(B'), hash(C')}